File: Dopamine.CollectionModule/ViewModels/CollectionViewModel.vb
Imports Dopamine.Core.Prism
Imports Microsoft.Practices.Prism.Commands
Imports Microsoft.Practices.Prism.Mvvm
Imports Microsoft.Practices.Prism.Regions

Namespace ViewModels
    Public Class CollectionViewModel
        Inherits BindableBase

#Region "Variables"
        Private ReadOnly mRegionManager As IRegionManager
        Private mSlideInFrom As Integer
        Private mPreviousIndex As Integer = 0
#End Region

#Region "Properties"
        Public NavigateBetweenCollectionCommand As DelegateCommand(Of String)

        Public Property SlideInFrom() As Integer
            Get
                Return mSlideInFrom
            End Get
            Set(ByVal value As Integer)
                SetProperty(Of Integer)(mSlideInFrom, value)
            End Set
        End Property
#End Region

#Region "Construction"
        Public Sub New(iRegionManager As IRegionManager)
            Me.mRegionManager = iRegionManager
            Me.NavigateBetweenCollectionCommand = New DelegateCommand(Of String)(AddressOf NavigateBetweenCollection)
            ApplicationCommands.NavigateBetweenCollectionCommand.RegisterCommand(Me.NavigateBetweenCollectionCommand)
            Me.SlideInFrom = 30
        End Sub
#End Region

#Region "Functions"
        Private Sub NavigateBetweenCollection(iIndex As String)

            If String.IsNullOrWhiteSpace(iIndex) Then Return

            Dim index As Integer = 0

            Integer.TryParse(iIndex, index)

            If index = 0 Then Return

            Me.SlideInFrom = If(index <= mPreviousIndex, -30, 30)

            mPreviousIndex = index

            mRegionManager.RequestNavigate(RegionNames.CollectionContentRegion, Me.GetPageForIndex(index))
        End Sub

        Private Function GetPageForIndex(iIndex As Integer) As String

            Dim page As String = String.Empty

            Select Case iIndex
                Case 1
                    page = GetType(Views.CollectionArtists).FullName
                Case 2
                    page = GetType(Views.CollectionGenres).FullName
                Case 3
                    page = GetType(Views.CollectionAlbums).FullName
                Case 4
                    page = GetType(Views.CollectionTracks).FullName
                Case 5
                    page = GetType(Views.CollectionPlaylists).FullName
                Case 6
                    page = GetType(Views.CollectionCloud).FullName
                Case Else
                    page = GetType(Views.CollectionArtists).FullName
            End Select

            Return page
        End Function
#End Region
    End Class
End Namespace

File: Dopamine.CollectionModule/ViewModels/CollectionViewModel.cs
using Dopamine.Core.Prism;
using Microsoft.Practices.Prism.Commands;
using Microsoft.Practices.Prism.Mvvm;
using Microsoft.Practices.Prism.Regions;

namespace Dopamine.CollectionModule.ViewModels
{
    public class CollectionViewModel : BindableBase
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private int slideInFrom;
        private int previousIndex = 0;
        #endregion

        #region Properties
        public DelegateCommand<string> NavigateBetweenCollectionCommand;

        public int SlideInFrom
        {
            get { return this.slideInFrom; }
            set { SetProperty<int>(ref this.slideInFrom, value); }
        }
        #endregion

        #region Construction
        public CollectionViewModel(IRegionManager regionManager)
        {
            this.regionManager = regionManager;
            this.NavigateBetweenCollectionCommand = new DelegateCommand<string>(NavigateBetweenCollection);
            ApplicationCommands.NavigateBetweenCollectionCommand.RegisterCommand(this.NavigateBetweenCollectionCommand);
            this.SlideInFrom = 30;
        }
        #endregion

        #region Functions
        private void NavigateBetweenCollection(string indexString)
        {
            if (string.IsNullOrWhiteSpace(indexString))
                return;

            int index = 0;

            int.TryParse(indexString, out index);

            if (index == 0)
                return;

            this.SlideInFrom = index <= this.previousIndex ? -30 : 30;

            this.previousIndex = index;

            this.regionManager.RequestNavigate(RegionNames.CollectionContentRegion, this.GetPageForIndex(index));
        }

        private string GetPageForIndex(int iIndex)
        {
            string page = string.Empty;

            switch (iIndex)
            {
                case 1:
                    page = typeof(Views.CollectionArtists).FullName;
                    break;
                case 2:
                    page = typeof(Views.CollectionGenres).FullName;
                    break;
                case 3:
                    page = typeof(Views.CollectionAlbums).FullName;
                    break;
                case 4:
                    page = typeof(Views.CollectionTracks).FullName;
                    break;
                case 5:
                    page = typeof(Views.CollectionPlaylists).FullName;
                    break;
                case 6:
                    page = typeof(Views.CollectionCloud).FullName;
                    break;
                default:
                    page = typeof(Views.CollectionArtists).FullName;
                    break;
            }

            return page;
        }
        #endregion
    }
}
